content/blog/2018-12-28-subquery.md

```diff
@@ -0,0 +1,49 @@
+---
+title: Subquery
+created_at: 2018-12-28
+kind: article
+author_name: Ganesh Vernekar
+---
+
+## Introduction
+
+As the title suggests, subquery is a part of a query, and allows you to do a range query within a query, which was not possible before. It has been a long standing request on Prometheus [prometheus/prometheus/1227](https://github.com/prometheus/prometheus/issues/1227) and a desired feature.
+
+PR for subquery was recently merged into Prometheus [prometheus/prometheus/4831](https://github.com/prometheus/prometheus/pull/4831), and will be available in Prometheus 2.7 release. Let’s learn more about it below.
+
+## Motivation
+
+Previously if you wanted to have a range selection on a query for your alerting rule or graphing, it would require you to have a recording rule based on that query, and perform range selection on the recording rule.
```

Comment:
You can't perform range selection on a recording rule, it's on the metrics created by the rule.